tests/FSharp.Compiler.ComponentTests/Language/DotLambdaTests.fs

```diff
@@ -0,0 +1,95 @@
+
+namespace Language
+
+open Xunit
+open FSharp.Test.Compiler
+
+module DotLambdaTests =
+
+    [<Fact>]
+    let ``DotLambda does NOT generalize automatically to a member based SRTP`` () =
+        Fsx "let inline myFunc x = x |> _.WhatANiceProperty"
+        |> withLangVersionPreview
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [(Error 72, Line 1, Col 28, Line 1, Col 47, "Lookup on object of indeterminate type based on information prior to this program point. A type annotation may be needed prior to this program point to constrain the type of the object. This may allow the lookup to be resolved.")] 
+
+    [<Fact>]
+    let ``DotLambda does allow member based SRTP if labelled explicitely`` () =
+        Fsx "let inline myFunc<'a when 'a:(member WhatANiceProperty: int)> (x: 'a) = x |> _.WhatANiceProperty "
+        |> withLangVersionPreview
+        |> typecheck
+        |> shouldSucceed    
+
+    [<Fact>]
+    let ``ToString with preview version`` () =
+        Fsx "let myFunc = _.ToString()"
+        |> withLangVersionPreview
+        |> typecheck
+        |> shouldSucceed
+
+    [<Fact>]
+    let ``Regression in neg typecheck hole as left arg`` () =
+        Fsx """
+let a = ( upcast _ ) : obj
+let b = ( _ :> _ ) : obj
+let c = ( _ :> obj) """
+        |> withLangVersionPreview
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            Error 10, Line 2, Col 20, Line 2, Col 21, "Unexpected symbol ')' in expression. Expected '.' or other token."
+            Error 10, Line 3, Col 13, Line 3, Col 15, "Unexpected symbol ':>' in expression. Expected '.' or other token."
+            Error 583, Line 3, Col 9, Line 3, Col 10, "Unmatched '('"
+            Error 10, Line 4, Col 13, Line 4, Col 15, "Unexpected symbol ':>' in expression. Expected '.' or other token."
+            Error 583, Line 4, Col 9, Line 4, Col 10, "Unmatched '('"]
+        
+    [<Fact>]
+    let ``ToString with F# 7`` () =
+        Fsx "_.ToString()"
+        |> withLangVersion70
+        |> typecheck
+        |> shouldFail
+        |> withSingleDiagnostic (Error 3350, Line 1, Col 1, Line 1, Col 3, "Feature 'underscore dot shorthand for accessor only function' is not available in F# 7.0. Please use language version 'PREVIEW' or greater." )
+        
+    [<Fact>]
+    let ``Simple anonymous unary function shorthands compile`` () =
+        FSharp """
+module One
+let a1 : {| Foo : {| Bar : string |}|} -> string = _.Foo.Bar
+let a2 : {| Foo : int array |} -> int = _.Foo.[5]
+let a3 : {| Foo : int array |} -> int = _.Foo[5]
+let a4 : {| Foo : unit -> string |} -> string = _.Foo()
+let a5 : {| Foo : int -> {| X : string |} |} -> string = _.Foo(5).X
+
+open System
+let a6 = [1] |> List.map _.ToString()
+        """
+        |> withLangVersionPreview
+        |> typecheck
+        |> shouldSucceed
+        
+    [<Fact>]
+    let ``Nested anonymous unary function shorthands compile`` () =
```

Comment:
Should it compile as the name of the test suggests? Or should it fail to compile as it does in the test?

Reply:
Sorry, I only noticed now this is still(/again) a draft

Reply:
It should fail, test is to be updated